libakumuli/storage_engine/blockstore.h

```diff
@@ -117,37 +117,72 @@ class FixedSizeFileStorage : public BlockStore,
     std::vector<std::string> volume_names_;
 
     //! Secret c-tor.
-    FixedSizeFileStorage(std::string metapath, std::vector<std::string> volpaths);
+    FileStorage(std::string metapath, std::vector<std::string> volpaths);
 
+    virtual void update_current_volume() = 0;
```

Comment:
My take on this is something like `adjust_current_volume` or `handle_volume_transition`.